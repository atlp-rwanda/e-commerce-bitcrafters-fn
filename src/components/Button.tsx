import React from 'react';

interface InputProps {
  name?: string
  value?: string;
  color?: string | any;
  rounded?: string;
  icon?: any;
  reverse?: boolean;
  textColor?: string | any;
  onClick?: (event: any) => void;
  disabled?: boolean
  type?: "submit" | "reset" | "button" | undefined
}

const Button: React.FC<InputProps> = (props) => {
// const [showPassword, setShowPassword] = useState(false)

  return (
    <button type={props.type} style={{backgroundColor:props.color || "rgb(38 38 38)",display:"flex", flexDirection:props.reverse?"row-reverse" :"row"}}  className={` rounded-sm mt-2 px-4 w-full py-3 border-none flex space-x-2 items-center justify-center gap-2`} onClick={props.onClick}>
 {props.icon} 

 <p className={` text-sm`} style={{color:props.textColor || "white"}}>{props.value}</p>

    </button>
  );
};

export default Button;