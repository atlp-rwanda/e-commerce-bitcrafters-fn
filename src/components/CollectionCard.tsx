import React from 'react';


interface InputProps {
  name?: string;
  Image?: string | any;
}

const CollectionCard: React.FC<InputProps> = (props) => {

  return (
<div className='flex gap-2 items-center justify-start my-3  flex-col m-1 mb-4 group  '>
<div className=" flex flex-col gap-3 items-center justify-center ">
      <img src={props.Image} alt={props.name} className="flex border border-gray_100 rounded-md w-44 h-44 object-contain"/>
      <p className="text-center">{props.name}</p>
    </div>
</div>
  );
};

export default CollectionCard;