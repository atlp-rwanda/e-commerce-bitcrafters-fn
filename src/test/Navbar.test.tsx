// Navbar.test.tsx

import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Navbar from "../components/Navbar";
import "@testing-library/jest-dom";
import { getDashboardLink } from "../components/Navbar";

const mockStore = configureStore([]);

jest.mock("../assets/images/Bit.Shop.svg", () => "mocked-logo.svg");

describe("Navbar Component", () => {
  let store: any;

  beforeEach(() => {
    store = mockStore({
      auth: { isLoggedIn: false },
      chat: { unreadMessagesCount: 0 },
      cart: { count: 0 },
      wishList: { count: 0 },
      notifications: {
    unreadCount: 5,
    isOpen: false,
    notifications: [
      {
        id: 1,
        message: "New product available",
        timestamp: "2024-07-18T10:00:00Z",
        isRead: false
      },
      {
        id: 2,
        message: "Your order has been shipped",
        timestamp: "2024-07-17T15:30:00Z",
        isRead: true
      },
    ]
  }
    });
  });

  const renderWithRouter = (component: React.ReactNode) => {
    return render(
      <Provider store={store}>
        <BrowserRouter>{component}</BrowserRouter>
      </Provider>
    );
  };

  test("renders logo", () => {
    renderWithRouter(<Navbar />);
    const logoElement = screen.getByAltText("Logo");
    expect(logoElement).toBeInTheDocument();
    expect(logoElement).toHaveAttribute("src", "mocked-logo.svg");
  });

  test("renders navigation links on desktop", () => {
    renderWithRouter(<Navbar />);
    const homeLink = screen.getByText("Home");
    const aboutLink = screen.getByText("About");
    const shopLink = screen.getByText("Shop");
    const contactLink = screen.getByText("Contact");

    expect(homeLink).toBeInTheDocument();
    expect(aboutLink).toBeInTheDocument();
    expect(shopLink).toBeInTheDocument();
    expect(contactLink).toBeInTheDocument();
  });

  test("renders login button when not logged in", () => {
    renderWithRouter(<Navbar />);
    const loginButtons = screen.getAllByText("Log in");
    expect(loginButtons.length).toBeGreaterThan(0);
  });

  test("renders search container", () => {
    renderWithRouter(<Navbar />);
    const searchContainer = screen.getByText("search here");
    expect(searchContainer).toBeInTheDocument();
  });

  test("toggles search component when search icon is clicked", () => {
    renderWithRouter(<Navbar />);
    const searchIcon = screen.getAllByRole('img', { hidden: true })[0];
    fireEvent.click(searchIcon);
    const searchContainer = screen.getByText("search here");
    expect(searchContainer).toBeInTheDocument();
  });

test("renders user icons when logged in", () => {
  store = mockStore({
    auth: { isLoggedIn: true },
    chat: { unreadMessagesCount: 0 },
    cart: { count: 2 },
    wishList: { count: 1 },
  });

  renderWithRouter(<Navbar />);

  const heartIcon = screen.getByLabelText('Heart');
  expect(heartIcon).toBeInTheDocument();
  const  cartIcon = screen.getByLabelText('Cart');
  expect(cartIcon).toBeInTheDocument();
});

test("opens chat when chat button is clicked", () => {
  store = mockStore({
    auth: { isLoggedIn: true },
    chat: { unreadMessagesCount: 0 },
    cart: { count: 0 },
    wishList: { count: 0 },
  });

  renderWithRouter(<Navbar />);

  const chatButton = screen.getByTestId("chat-button");
  fireEvent.click(chatButton);

  const chatContainer = screen.getByTestId("chat-container");
  expect(chatContainer).toBeInTheDocument();
});
});
describe("Navbar Component Additional Tests", () => {
  let store: any;

  beforeEach(() => {
    store = mockStore({
      auth: { isLoggedIn: true, authRole: "user" },
      chat: { unreadMessagesCount: 2 },
      cart: { count: 3 },
      wishList: { count: 1 },
    });
  });

  const renderWithRouter = (component: React.ReactNode) => {
    return render(
      <Provider store={store}>
        <BrowserRouter>{component}</BrowserRouter>
      </Provider>
    );
  };

  test("renders wishlist link when logged in", () => {
    renderWithRouter(<Navbar />);
    const wishlistLink = screen.getByLabelText("Heart");
    expect(wishlistLink).toBeInTheDocument();
    expect(wishlistLink.closest('a')).toHaveAttribute("href", "/wishList");
  });

  test("renders cart link with correct count when logged in", () => {
    renderWithRouter(<Navbar />);
    const cartLink = screen.getByLabelText("Cart");
    expect(cartLink).toBeInTheDocument();
    expect(cartLink.closest('a')).toHaveAttribute("href", "/cart");
    const cartCount = screen.getByText("3");
    expect(cartCount).toBeInTheDocument();
  });

  test("renders orders link when logged in", () => {
    renderWithRouter(<Navbar />);
    const ordersLink = screen.getByTestId("orders-link");
    expect(ordersLink).toBeInTheDocument();
    expect(ordersLink).toHaveAttribute("href", "/orders");
  });

  test("renders burger menu on mobile when logged in", () => {
    renderWithRouter(<Navbar />);
    const burgerButton = screen.getByTestId("burger-button");
    expect(burgerButton).toBeInTheDocument();
    fireEvent.click(burgerButton);
    const mobileMenu = screen.getByTestId("mobile-menu");
    expect(mobileMenu).toBeInTheDocument();
  });

  test("renders dashboard link in mobile menu for seller", () => {
    store = mockStore({
      auth: { isLoggedIn: true, authRole: "seller" },
      chat: { unreadMessagesCount: 0 },
      cart: { count: 0 },
      wishList: { count: 0 },
    });
    renderWithRouter(<Navbar />);
    const burgerButton = screen.getByTestId("burger-button");
    fireEvent.click(burgerButton);
    
    const dashboardLink = screen.getByText("Dashboard");
    expect(dashboardLink).toBeInTheDocument();
    expect(dashboardLink.closest('a')).toHaveAttribute("href", "/seller");
  });

  test("renders dashboard link in mobile menu for admin", () => {
    store = mockStore({
      auth: { isLoggedIn: true, authRole: "admin" },
      chat: { unreadMessagesCount: 0 },
      cart: { count: 0 },
      wishList: { count: 0 },
    });
    renderWithRouter(<Navbar />);
    const burgerButton = screen.getByTestId("burger-button");
    fireEvent.click(burgerButton);
    
    const dashboardLink = screen.getByText("Dashboard");
    expect(dashboardLink).toBeInTheDocument();
    expect(dashboardLink.closest('a')).toHaveAttribute("href", "/admin");
  });

  test("renders logout link in mobile menu", () => {
    renderWithRouter(<Navbar />);
    const burgerButton = screen.getByTestId("burger-button");
    fireEvent.click(burgerButton);
    
    const logoutLink = screen.getByText("Logout");
    expect(logoutLink).toBeInTheDocument();
  });

  test("closes mobile menu when a link is clicked", () => {
    renderWithRouter(<Navbar />);
    const burgerButton = screen.getByTestId("burger-button");
    fireEvent.click(burgerButton);
    
    const profileLink = screen.getByText("Profile");
    fireEvent.click(profileLink);
    
    const mobileMenu = screen.queryByTestId("mobile-menu");
    expect(mobileMenu).not.toBeInTheDocument();
  });

test("closes profile modal when close button is clicked", () => {
    renderWithRouter(<Navbar />);
    const profileButton = screen.getByTestId("profile-button");
    fireEvent.click(profileButton);
    const profileModal = screen.getByText("User settings");
    expect(profileModal).toBeInTheDocument();
 const closeButton = screen.getByTestId("close-button");
    fireEvent.click(closeButton);
    expect(profileModal).not.toBeInTheDocument();
  });
});
describe("getDashboardLink function", () => {
  test("returns '/seller' for seller role", () => {
    expect(getDashboardLink("seller")).toBe("/seller");
  });

  test("returns '/admin' for admin role", () => {
    expect(getDashboardLink("admin")).toBe("/admin");
  });

  test("returns '/profile' for user role", () => {
    expect(getDashboardLink("user")).toBe("/profile");
  });

  test("returns '/profile' for any other role", () => {
    expect(getDashboardLink("unknown")).toBe("/profile");
  });
});