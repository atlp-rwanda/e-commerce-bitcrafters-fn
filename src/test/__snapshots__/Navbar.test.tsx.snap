// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders Footer correctly 1`] = `
<footer
  className="bg-black w-full "
>
  <div
    className="footer bg-black w-full p-5 px-10 flex flex-row flex-wrap justify-between flex-1 items-start py-12"
  >
    <div
      className="items-start justify-center flex flex-col w-[90%] mb-10 tablet:mb-0 tablet:w-[30%] space-y-4"
    >
      <img
        alt="logo"
        className="r w-24"
      />
      <p
        className="text-xs text-gray-400 text-start w-[70%]"
      >
        We have your satisfaction in mind Lorem ipsum dolor sit amet consectetur. In eu aliquet orci ac 
      </p>
      <div
        className="flex-between space-x-2"
      >
        <div
          className="border p-1 border-gray_100 rounded-full"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="LinkedIn"
            >
              <g>
                <path
                  d="M18.44,3.06H5.56a2.507,2.507,0,0,0-2.5,2.5V18.44a2.507,2.507,0,0,0,2.5,2.5H18.44a2.5,2.5,0,0,0,2.5-2.5V5.56A2.5,2.5,0,0,0,18.44,3.06Zm1.5,15.38a1.511,1.511,0,0,1-1.5,1.5H5.56a1.511,1.511,0,0,1-1.5-1.5V5.56a1.511,1.511,0,0,1,1.5-1.5H18.44a1.511,1.511,0,0,1,1.5,1.5Z"
                />
                <g>
                  <path
                    d="M6.376,10.748a1,1,0,1,1,2,0v6.5h0a1,1,0,0,1-2,0Z"
                  />
                  <circle
                    cx="7.376"
                    cy="6.744"
                    r="1"
                  />
                  <path
                    d="M17.62,13.37v3.88a1,1,0,1,1-2,0V13.37a1.615,1.615,0,1,0-3.23,0v3.88a1,1,0,0,1-2,0v-6.5a1.016,1.016,0,0,1,1-1,.94.94,0,0,1,.84.47,3.609,3.609,0,0,1,5.39,3.15Z"
                  />
                </g>
              </g>
            </g>
          </svg>
        </div>
        <div
          className="border p-1 border-gray_100 rounded-full"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </div>
        <div
          className="border p-1 border-gray_100 rounded-full"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Twitter"
            >
              <path
                d="M19.913,5.322a1.034,1.034,0,0,1,.837,1.629L19.708,8.432c-.064,5.086-1.765,8.539-5.056,10.264A10.917,10.917,0,0,1,9.6,19.835a12.233,12.233,0,0,1-6.2-1.524.76.76,0,0,1-.317-.8.768.768,0,0,1,.63-.6,20.6,20.6,0,0,0,3.745-.886C2,13.5,3.19,7.824,3.71,6.081a1.028,1.028,0,0,1,1.729-.422,9.931,9.931,0,0,0,5.995,2.95A4.188,4.188,0,0,1,12.725,5.3a4.125,4.125,0,0,1,5.7.02ZM4.521,17.794c1.862.872,6.226,1.819,9.667.016,2.955-1.549,4.476-4.732,4.521-9.461a.771.771,0,0,1,.142-.436l1.081-1.538-.041-.053c-.518-.007-1.029-.014-1.55,0a.835.835,0,0,1-.547-.221,3.13,3.13,0,0,0-4.383-.072,3.174,3.174,0,0,0-.935,2.87.646.646,0,0,1-.154.545.591.591,0,0,1-.516.205A10.924,10.924,0,0,1,4.722,6.354c-.67,2.078-1.52,7.094,3.869,9.065a.632.632,0,0,1,.416.538.625.625,0,0,1-.3.6A13.178,13.178,0,0,1,4.521,17.794ZM11.875,8.65h0Zm7.793-.161,0,0Z"
              />
            </g>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="items-start justify-center flex flex-col w-[40%] bigphone:w-[12%] space-y-2 mb-10 phone:mb-0"
    >
      <p
        className="text-gray-200 font-semibold text-sm text-white"
      >
        Account
      </p>
      <p
        className=" text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          My Account
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Login / Register
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Cart
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Wishlist
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Shop
        </a>
      </p>
    </div>
    <div
      className="items-start justify-center flex flex-col w-[40%] bigphone:w-[12%] space-y-2 mb-10 phone:mb-0"
    >
      <p
        className="text-white font-semibold text-sm"
      >
        Information
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/FAQ"
          onClick={[Function]}
        >
          FAQ
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/about"
          onClick={[Function]}
        >
          About Us
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Resources
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Terms
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/privacyPolicy"
          onClick={[Function]}
        >
          Privacy Policy
        </a>
      </p>
    </div>
    <div
      className="items-start justify-center flex flex-col w-[80%] bigphone:w-[25%]  mt-4 bigphone:mt-1 space-y-2 mb-10 phone:mb-0"
    >
      <p
        className="text-white font-semibold text-sm"
      >
        Contact Us
      </p>
      <div
        className="flex-between items-start space-x-2"
      >
        <svg
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "white",
            }
          }
          viewBox="0 0 448 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M380.9 97.1C339 55.1 283.2 32 223.9 32c-122.4 0-222 99.6-222 222 0 39.1 10.2 77.3 29.6 111L0 480l117.7-30.9c32.4 17.7 68.9 27 106.1 27h.1c122.3 0 224.1-99.6 224.1-222 0-59.3-25.2-115-67.1-157zm-157 341.6c-33.2 0-65.7-8.9-94-25.7l-6.7-4-69.8 18.3L72 359.2l-4.4-7c-18.5-29.4-28.2-63.3-28.2-98.2 0-101.7 82.8-184.5 184.6-184.5 49.3 0 95.6 19.2 130.4 54.1 34.8 34.9 56.2 81.2 56.1 130.5 0 101.8-84.9 184.6-186.6 184.6zm101.2-138.2c-5.5-2.8-32.8-16.2-37.9-18-5.1-1.9-8.8-2.8-12.5 2.8-3.7 5.6-14.3 18-17.6 21.8-3.2 3.7-6.5 4.2-12 1.4-32.6-16.3-54-29.1-75.5-66-5.7-9.8 5.7-9.1 16.3-30.3 1.8-3.7.9-6.9-.5-9.7-1.4-2.8-12.5-30.1-17.1-41.2-4.5-10.8-9.1-9.3-12.5-9.5-3.2-.2-6.9-.2-10.6-.2-3.7 0-9.7 1.4-14.8 6.9-5.1 5.6-19.4 19-19.4 46.3 0 27.3 19.9 53.7 22.6 57.4 2.8 3.7 39.1 59.7 94.8 83.8 35.2 15.2 49 16.5 66.6 13.9 10.7-1.6 32.8-13.4 37.4-26.4 4.6-13 4.6-24.1 3.2-26.4-1.3-2.5-5-3.9-10.5-6.6z"
          />
        </svg>
        <div>
          <p
            className="text-xs text-gray-200 text-start text-white"
          >
            Whatsapp
          </p>
          <p
            className="text-xs text-gray-400 text-start hover:text-gray-200 transition-all "
          >
            <a
              href="/"
              onClick={[Function]}
            >
              +250 788 888 888
            </a>
          </p>
        </div>
      </div>
      <div
        className="flex-between items-start space-x-2"
      >
        <svg
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "white",
            }
          }
          viewBox="0 0 512 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M164.9 24.6c-7.7-18.6-28-28.5-47.4-23.2l-88 24C12.1 30.2 0 46 0 64C0 311.4 200.6 512 448 512c18 0 33.8-12.1 38.6-29.5l24-88c5.3-19.4-4.6-39.7-23.2-47.4l-96-40c-16.3-6.8-35.2-2.1-46.3 11.6L304.7 368C234.3 334.7 177.3 277.7 144 207.3L193.3 167c13.7-11.2 18.4-30 11.6-46.3l-40-96z"
          />
        </svg>
        <div>
          <p
            className="text-xs text-gray-200 text-start  text-white"
          >
            Call Us
          </p>
          <p
            className="text-xs text-gray-400 text-start hover:text-gray-200 transition-all "
          >
            <a
              href="tel:+250 788 888 888"
            >
              +250 788 888 888
            </a>
          </p>
        </div>
      </div>
      <div
        className="flex-between items-start space-x-2"
      >
        <svg
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
          />
        </svg>
        <div>
          <p
            className="text-xs text-gray-200 text-start  text-white"
          >
            Email Us
          </p>
          <p
            className="text-[10px] tablet:text-xs text-gray-400 text-start hover:text-gray-200 transition-all "
          >
            <a
              href="/bitcrafter.andela@gmail.com"
              onClick={[Function]}
            >
              bitcrafter.andela@gmail.com
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="p-6 flex items-center w-[80%] m-auto justify-center border-t-[1px] border-gray_100"
  >
    <p
      className="text-xs text-gray-400"
    >
      © 
      2024
       All rights reserved - Bitcrafters Andela Team
    </p>
  </div>
</footer>
`;

exports[`Navbar component renders Navbar correctly 1`] = `
<nav
  className="navbar-container  w-full border-b-1 p-8 flex-between space-x-2"
>
  <div
    className="nav-logo w-[20%] "
  >
    <img
      alt="Logo"
      className="w-24 object-contain "
    />
  </div>
  <div
    className="nav-link flex-between  w-[30%] hidden tablet:flex"
  >
    <ul
      className="flex-between space-x-4"
    >
      <li>
        <a
          className="text-sm"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="text-sm"
          href="/"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          className="text-sm"
          href="/"
          onClick={[Function]}
        >
          Shop
        </a>
      </li>
      <li>
        <a
          className="text-sm"
          href="/"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
  <div
    className="nav-icons flex-between hidden tablet:flex"
  >
    <ul
      className="flex-between space-x-4"
    >
      <li>
        <div
          className="search-container bg-gray rounded-sm flex-between space-x-2 p-2"
        >
          <p
            className="text-xs"
          >
            search here
          </p>
          <svg
            fill="currentColor"
            height="1em"
            id="search"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            version="1.1"
            viewBox="0 0 24 24"
            width="1em"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g>
              <path
                d="M20.031,20.79c0.46,0.46,1.17-0.25,0.71-0.7l-3.75-3.76c1.27-1.41,2.04-3.27,2.04-5.31
		c0-4.39-3.57-7.96-7.96-7.96s-7.96,3.57-7.96,7.96c0,4.39,3.57,7.96,7.96,7.96c1.98,0,3.81-0.73,5.21-1.94L20.031,20.79z
		 M4.11,11.02c0-3.84,3.13-6.96,6.96-6.96c3.84,0,6.96,3.12,6.96,6.96c0,3.84-3.12,6.96-6.96,6.96C7.24,17.98,4.11,14.86,4.11,11.02
		z"
              />
            </g>
          </svg>
        </div>
      </li>
      <li>
        <a
          className="text-lg"
          href="/"
          onClick={[Function]}
        >
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Heart"
            >
              <path
                d="M12,20.043a.977.977,0,0,1-.7-.288L4.63,13.08A5.343,5.343,0,0,1,6.053,4.513,5.266,5.266,0,0,1,12,5.371a5.272,5.272,0,0,1,5.947-.858A5.343,5.343,0,0,1,19.37,13.08l-6.676,6.675A.977.977,0,0,1,12,20.043ZM8.355,4.963A4.015,4.015,0,0,0,6.511,5.4,4.4,4.4,0,0,0,4.122,8.643a4.345,4.345,0,0,0,1.215,3.73l6.675,6.675,6.651-6.675a4.345,4.345,0,0,0,1.215-3.73A4.4,4.4,0,0,0,17.489,5.4a4.338,4.338,0,0,0-4.968.852h0a.744.744,0,0,1-1.042,0A4.474,4.474,0,0,0,8.355,4.963Z"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          className="text-lg"
          href="/"
          onClick={[Function]}
        >
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="176"
              cy="416"
              fill="none"
              r="16"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
            <circle
              cx="400"
              cy="416"
              fill="none"
              r="16"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
            <path
              d="M48 80h64l48 272h256"
              fill="none"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
            <path
              d="M160 288h249.44a8 8 0 0 0 7.85-6.43l28.8-144a8 8 0 0 0-7.85-9.57H128"
              fill="none"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          className="text-lg"
          href="/"
          onClick={[Function]}
        >
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M427.68 351.43C402 320 383.87 304 383.87 217.35 383.87 138 343.35 109.73 310 96c-4.43-1.82-8.6-6-9.95-10.55C294.2 65.54 277.8 48 256 48s-38.21 17.55-44 37.47c-1.35 4.6-5.52 8.71-9.95 10.53-33.39 13.75-73.87 41.92-73.87 121.35C128.13 304 110 320 84.32 351.43 73.68 364.45 83 384 101.61 384h308.88c18.51 0 27.77-19.61 17.19-32.57zM320 384v16a64 64 0 0 1-128 0v-16"
              fill="none"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
  <div
    className=" w-[5%] hidden tablet:flex"
  >
    <a
      className="text-lg"
      href="/login"
      onClick={[Function]}
    >
       
      <svg
        className="text-2xl"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M0.877014 7.49988C0.877014 3.84219 3.84216 0.877045 7.49985 0.877045C11.1575 0.877045 14.1227 3.84219 14.1227 7.49988C14.1227 11.1575 11.1575 14.1227 7.49985 14.1227C3.84216 14.1227 0.877014 11.1575 0.877014 7.49988ZM7.49985 1.82704C4.36683 1.82704 1.82701 4.36686 1.82701 7.49988C1.82701 8.97196 2.38774 10.3131 3.30727 11.3213C4.19074 9.94119 5.73818 9.02499 7.50023 9.02499C9.26206 9.02499 10.8093 9.94097 11.6929 11.3208C12.6121 10.3127 13.1727 8.97172 13.1727 7.49988C13.1727 4.36686 10.6328 1.82704 7.49985 1.82704ZM10.9818 11.9787C10.2839 10.7795 8.9857 9.97499 7.50023 9.97499C6.01458 9.97499 4.71624 10.7797 4.01845 11.9791C4.97952 12.7272 6.18765 13.1727 7.49985 13.1727C8.81227 13.1727 10.0206 12.727 10.9818 11.9787ZM5.14999 6.50487C5.14999 5.207 6.20212 4.15487 7.49999 4.15487C8.79786 4.15487 9.84999 5.207 9.84999 6.50487C9.84999 7.80274 8.79786 8.85487 7.49999 8.85487C6.20212 8.85487 5.14999 7.80274 5.14999 6.50487ZM7.49999 5.10487C6.72679 5.10487 6.09999 5.73167 6.09999 6.50487C6.09999 7.27807 6.72679 7.90487 7.49999 7.90487C8.27319 7.90487 8.89999 7.27807 8.89999 6.50487C8.89999 5.73167 8.27319 5.10487 7.49999 5.10487Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </a>
  </div>
  <button
    className="tablet:hidden"
    onClick={[Function]}
  >
    <svg
      className="text-3xl "
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 15 15"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </button>
</nav>
`;
